hera_cal/lst_stack/config.py

```diff
@@ -386,103 +256,644 @@ def make_lst_bin_config_file(
+            autos=[(a, b) for a, b in autos],
+            pols=[p.decode() for p in pols],
+            calfiles=calfiles,
+            inpaint_files=inpaint_files,
+        )
+
+    def _get_single_config(self, outfile, lstindex: int | None) -> LSTConfigSingle:
+        """Return a single LSTConfigSingle object.
+
+        This method is used to create a LSTConfigSingle object from the current
+        LSTConfig object. It is used by the at_single_outfile and at_single_bin
+        methods. The output LSTConfigSingle object represents the files and config
+        required for a single LST bin or output file (which might contain a few
+        LST bins).
+        """
+        lst_grid = self.lst_grid[outfile]
+        grid_edges = self.lst_grid_edges[outfile]
+
+        if lstindex is not None:
+            lst_grid = lst_grid[lstindex]
+            grid_edges = grid_edges[lstindex:lstindex + 2]
+
+        tinds, _, matched_files, cals, inp = filter_required_files_by_times(
+            lst_range=(grid_edges[0], grid_edges[-1]),
+            data_metas=self.matched_metas[outfile],
+            cal_files=self.calfiles[outfile] if self.calfiles else None,
+            where_inpainted_files=self.inpaint_files[outfile] if self.inpaint_files else None,
+        )
+
+        kw = attrs.asdict(self, recurse=False)
+        kw['lst_grid'] = lst_grid
+        kw['matched_files'] = [m.path for m in matched_files]
+        kw['calfiles'] = cals
+        kw['inpaint_files'] = inp
+
+        return LSTConfigSingle(time_indices=tinds, **kw)
+
+    def at_single_outfile(
+        self,
+        outfile: int | None = None,
+        lst: float | None = None,
+    ) -> LSTConfigSingle:
+        if lst is None and outfile is None:
+            raise ValueError("Either lst or outfile must be specified.")
+
+        if lst is not None and outfile is not None:
+            raise ValueError("Only one of lst or outfile can be specified.")
+
+        if lst is not None:
+            outfile = np.searchsorted(self.lst_grid_edges[:, 0], lst, side='right') - 1
+
+        return self._get_single_config(outfile, None)
+
+    def at_single_bin(
+        self,
+        bin_index: int | None = None,
+        lst: float | None = None,
+    ) -> LSTConfigSingle:
+        if lst is None and bin_index is None:
+            raise ValueError("Either lst or bin_index must be specified.")
+
+        if lst is not None and bin_index is not None:
+            raise ValueError("Only one of lst or bin_index can be specified.")
+
+        edges = self.lst_grid_edges[:, :-1].flatten()
+
+        if lst is not None:
+            bin_index = np.searchsorted(edges, lst, side='right') - 1
+
+        fl_index = bin_index // self.config.nlsts_per_file
+        bin_index = bin_index % self.config.nlsts_per_file
+
+        return self._get_single_config(fl_index, bin_index)
+
+
+@attrs.define(slots=False, kw_only=True)
+class LSTConfigSingle(_LSTConfigBase):
```

Comment:
That was my original intention -- to use one class and a `select()`. And it could be done, however, there is a difference in the shape of the `matched_files` between the two: in the larger case, `matched_files` is a triple-nested list (outfile, night, files), whereas for the single it's a single list of files (the nights all get lumped together).

This means that validation of the parameters is slightly different in each case. Also, the single has some extra attributes, like `matched_lsts` which read the files to get the LSTs in each. Long story short, I ended up thinking it was cleaner to split the classes. I could be convinced otherwise though.

Perhaps one way to do this would be to not care about "nights" -- these are useful to keep separate in the `LSTBinConfigurator` (because it helps find calfiles etc), but I think maybe it's no longer useful in the `LSTConfig` -- you just care about which output file an input file belongs to. Still, to keep the shapes of the matched_files consistent after a select would mean that the resulting class would have matched_files that are a list of lists, where the outer list is length 1. I don't love that either, but maybe it's worth it for clearing up two whole classes...